File: [프로그래머스] 코딩테스트 고득점 Kit/해시/Level 1/폰켓몬.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/1845?language=cpp

#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums) {
    int max = nums.size() / 2;
    unordered_set<int> nums_set(nums.begin(), nums.end());
    
    if(nums_set.size() > max)
        return max;
    else
        return nums_set.size();
}

File: [프로그래머스] 코딩테스트 고득점 Kit/해시/Level 1/폰켓몬.java
// https://school.programmers.co.kr/learn/courses/30/lessons/1845?language=cpp

import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int max = nums.length / 2;
        HashSet<Integer> hashSet = new HashSet<>();

        for (int num : nums) {
            hashSet.add(num);
        }

        if(hashSet.size() > max)
            return max;
        else
            return hashSet.size();
    }
}